assets/src/edit-story/components/canvas/karma/pageMenuActions.karma.js

```diff
@@ -0,0 +1,123 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { waitFor } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../karma';
+import { useStory } from '../../../app/story';
+import { useInsertElement } from '../../../components/canvas';
+import { TEXT_ELEMENT_DEFAULT_FONT } from '../../../app/font/defaultFonts';
+
+describe('PageMenu integration', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+    await fixture.render();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  describe('add a text', () => {
+    let element;
+
+    beforeEach(async () => {
+      const insertElement = await fixture.renderHook(() => useInsertElement());
+      element = await fixture.act(() =>
+        insertElement('text', {
+          font: TEXT_ELEMENT_DEFAULT_FONT,
+          content: 'hello world!',
+          x: 40,
+          y: 40,
+          width: 250,
+        })
+      );
+      // @todo: The focusing is currently done via timeout. Find a way to
+      // make this nicer.
+      const framesLayer = fixture.querySelector('[data-testid="FramesLayer"]');
+      await waitFor(() => {
+        if (!framesLayer.contains(document.activeElement)) {
+          throw new Error('Focus is not set on the canvas yet');
+        }
+      });
+    });
+
+    function getFrame() {
+      return fixture.querySelector(
+        `[data-element-id="${element.id}"] [data-testid="textFrame"]`
+      );
+    }
+
+    async function getSelection() {
+      const storyContext = await fixture.renderHook(() => useStory());
+      return storyContext.state.selectedElementIds;
+    }
+
+    it('should render initial content and make it selected', async () => {
+      expect(getFrame().textContent).toEqual('hello world!');
+      expect(await getSelection()).toEqual([element.id]);
+    });
+
+    describe('delete the element, restore, and delete again', () => {
+      it('using shortcuts', async () => {
+        // Delete.
+        await fixture.events.keyboard.shortcut('del');
+        expect(getFrame()).toBeNull();
+        expect(await getSelection()).toEqual([]);
+
+        // Restore.
+        await fixture.events.keyboard.shortcut('mod+z');
+        expect(getFrame()).toBeTruthy();
+        expect(await getSelection()).toEqual([element.id]);
+
+        // Delete again.
+        await fixture.events.keyboard.shortcut('del');
+        expect(getFrame()).toBeNull();
+        expect(await getSelection()).toEqual([]);
+      });
+
+      it('using page menu', async () => {
+        // Delete.
+        await fixture.events.keyboard.shortcut('del');
+        expect(getFrame()).toBeNull();
+        expect(await getSelection()).toEqual([]);
+
+        // Restore.
+        // @todo: Use either `workspace.pageMenu.undoButton` or
+        // `byRole('button', {name: 'Undo Changes'})` APIs.
```

Comment:
Is this todo for later or for this PR?

Reply:
Yeah. I'm leaving the trail to do a follow up on all queries in the Karma tests.